### PR TITLE
feat(feature/cvsb-11782): As a VTM user, I want to be able to add a test, so that I am able to record a contingency test

### PR DESCRIPTION
**Pull request by irina005**
## As a VTM user, I want to be able to add a test, so that I am able to record a contingency test
https://jira.dvsacloud.uk/browse/CVSB-11782

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/store/effects/VehicleTestResult.effects.ts [hunk 1]
```diff
@@ -150,6 +152,26 @@ export class VehicleTestResultModelEffects {
     })
   );
 
+  @Effect()
+  createTestRecord$ = this._actions$.pipe(
+    ofType<CreateTestResult>(EVehicleTestResultActions.CreateTestResult),
+    map(({ vTestResultUpdated }): VehicleTestResultUpdate => vTestResultUpdated),
+    switchMap((vTestResultUpdated) => {
+      vTestResultUpdated.msUserDetails = { ...this.loggedUser.getUser() };
+      this.router.navigate(['/test-record', -1], {
```
Comment (JonLazariniDD) [inline src/app/store/effects/VehicleTestResult.effects.ts:161]: I will need to catch up on that one please.

File: src/app/store/reducers/VehicleTestResult.reducers.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        selectedTestResultModel: action.vTestResultUpdated
+      };
+    }
+
+    case EVehicleTestResultActions.CreateTestResultSuccess: {
+      return {
+        ...state,
+        selectedTestResultModel: action.vTestResultUpdated
+      };
+    }
```
Comment (alleypa) [inline src/app/store/reducers/VehicleTestResult.reducers.ts:67]: lets discuss on this. Do we need both to update the state

File: src/app/store/selectors/VehicleTestResult.selectors.ts [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+  VehicleTechRecordEdit,
+  VehicleTechRecordModel
+} from '@app/models/vehicle-tech-record.model';
+import { VrmModel } from '@app/models/vrm.model';
 import { TestRecordTestType } from '@app/models/test-record-test-type';
 import { TestType } from '@app/models/test.type';
-import { TestResultModel } from '@app/models/test-result.model';
 import { TestTypeCategory } from '@app/models/test-type-category';
 import { TreeData } from '@app/models/tree-data';
+import { VehicleTestResultUpdate } from '@app/models/vehicle-test-result-update';
```
Comment (alleypa) [inline src/app/store/selectors/VehicleTestResult.selectors.ts:14]: `VehicleTestResultUpdate, getActiveVehicleTechRecord, getSelectedVehicleTechRecord `
are u using these imports?

File: src/app/technical-record/technical-record.component.html [hunk 4]
```diff
@@ -6,27 +6,54 @@
...
-    </ng-container>
-
+        <div class="govuk-grid-column-one-third govuk-!-margin-left-0 govuk-!-padding-left-0">
+          <button
+            class="govuk-button govuk-button--secondary"
+            id="test-add-test"
+            type="button"
+            data-module="govuk-button"
+            [routerLink]="['/select-test-type', '-1']"
+          >
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.html:51]: what will -1 do

File: src/app/technical-record/technical-record.component.html [hunk 5]
```diff
@@ -6,27 +6,54 @@
...
+      </div>
 
-    <ng-container>
-      <button
-        id="test-save-btn"
-        class="govuk-button"
-        type="submit"
-        *ngIf="editState || createState"
+      <ng-container
+        *ngIf="
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.html:38]: lets pick up the discussion on this. Also we need to ask for the visibility of this button in edit & create mode? Same goes for the plate button when you rebase this branch. Should the plate button be visible in edit mode

File: src/app/technical-record/technical-record.container.ts [hunk 6]
```diff
@@ -79,4 +80,8 @@ export class TechnicalRecordsContainer implements OnInit {
   viewStateHandler(state: VIEW_STATE): void {
     this.store.dispatch(new SetViewState(state));
   }
+
+  selectTestTypeHandler(): void {
+    this.store.dispatch(new SelectTestType());
+  }
```
Comment (alleypa) [inline src/app/technical-record/technical-record.container.ts:86]: will you still need be needing `selectTestTypeHandler`

File: src/app/test-record/select-test-type/select-test-type.container.ts [hunk 7]
```diff
@@ -28,17 +39,38 @@ import { TreeData } from '@app/models/tree-data';
...
+    this.activeVehicleTechRecord$ = this.store
+      .select(getActiveVehicleTechRecord)
+      .pipe(map((f) => f()));
+    this.activeVehicleTechRecord$
+      .subscribe((result) => (this.activeRecord = result))
+      .unsubscribe();
+    this.filteredCategories$ = this.store.select(
+      getFilteredTestTypeCategories(this.activeRecord)
+    );
+    this.viewState$ = this.store.select(getTestViewState);
```
Comment (alleypa) [inline src/app/test-record/select-test-type/select-test-type.container.ts:66]: let discuss please

File: src/app/test-record/test-record.component.ts [hunk 8]
```diff
@@ -49,9 +49,10 @@ export class TestRecordComponent implements OnInit {
     this.hasDefectsApplicable = this.testTypesApplicable.defectsApplicable[
       this.testResultObj.testType.testTypeId
     ];
-    this.hasSeatBeltApplicable =
-      !(this.testTypesApplicable.seatBeltApplicable[this.testResultObj.testType.testTypeId] &&
-      this.testResultObj.testRecord.vehicleType === 'psv');
+    this.hasSeatBeltApplicable = !(
+      this.testTypesApplicable.seatBeltApplicable[this.testResultObj.testType.testTypeId] &&
+      this.testResultObj.testRecord.vehicleType === 'psv'
```
Comment (alleypa) [inline src/app/test-record/test-record.component.ts:54]: limit the use of ordinary string. Use `VEHICLE_TYPES.PSV`

File: src/app/test-record/select-test-type/select-test-type.container.ts [hunk 9]
```diff
@@ -28,17 +39,38 @@ import { TreeData } from '@app/models/tree-data';
 export class SelectTestTypeContainer implements OnInit {
   testRecord$: Observable<TestRecordTestType>;
   filteredCategories$: Observable<TreeData[]>;
+  viewState$: Observable<VIEW_STATE>;
+  activeVehicleTechRecord$: Observable<VehicleTechRecordEdit>;
+  activeRecord: VehicleTechRecordEdit;
```
Comment (alleypa) [inline src/app/test-record/select-test-type/select-test-type.container.ts:44]: `vehicleRecord: VehicleTechRecordEdit;`

**Pull request closed without merging**